diff --git a/src/publish/index.js b/src/publish/index.js
index 3f19167432b0c4fcc48afa4ba538ab054a4783c0..af2395a3dc2a266eb02fd1a3ad7dee918026d20d 100644
--- a/src/publish/index.js
+++ b/src/publish/index.js
@@ -25,7 +25,15 @@ import {
  * @returns {Promise<void>}
  */
 export const publish = async (options) => {
-  const { branchConfigs, packages, rootDir, branch, tag, ghToken } = options
+  const {
+    branchConfigs,
+    packages,
+    rootDir,
+    branch,
+    tag,
+    ghToken,
+    releaseAll = false,
+  } = options
 
   const branchName = /** @type {string} */ (branch ?? currentGitBranch())
   const isMainBranch = branchName === 'main'
@@ -65,9 +73,13 @@ export const publish = async (options) => {
   let range = `${latestTag}..HEAD`
   // let range = ``;
 
-  // If RELEASE_ALL is set via a commit subject or body, all packages will be
-  // released regardless if they have changed files matching the package srcDir.
-  let RELEASE_ALL = false
+  // All packages will be released if any of the following conditions are met:
+  // - RELEASE_ALL is set via a commit subject or body
+  // - A tag is explicitly set
+  // - The config option releaseAll is set to true
+  //
+  // Otherwise, only packages with changed files matching the package srcDir will be released.
+  let RELEASE_ALL = releaseAll
 
   if (!latestTag || tag) {
     if (tag) {
diff --git a/src/publish/types.d.ts b/src/publish/types.d.ts
index c241761cae5eb38e6c22c4ba3b9b87df1ce74c1c..5c346cbcfcd79c7584a9d2488e9561836dadb415 100644
--- a/src/publish/types.d.ts
+++ b/src/publish/types.d.ts
@@ -56,4 +56,6 @@ export type RunOptions = {
   tag?: string
   // The GitHub token used to search for user metadata and make a GitHub release.
   ghToken?: string
+  /** Release all packages.  Defaults to false but can be overridden with RELEASE_ALL in a commit message */
+  releaseAll?: boolean
 }
